agents-api/agents_api/routers/sessions/session.py

```diff
@@ -116,15 +115,18 @@ def _remove_messages(
 
         return result, token_count
 
-    def truncate(
-        self, messages: list[Entry], summarization_tokens_threshold: int
+    def _truncate_context(
```

Comment:
The only thing I wanted to check for truncation is that it should truncate from the top and not the bottom.

Meaning if we have messages: [m1,m2,m3…m25] then if there’s an overflow, drop m2, m3 and so on (not m1 if it’s a system message)

So the _truncate_context method should keep m1 if role==system and then start dropping from reversed(entries)